### PR TITLE
Define version in metadata and extract it using importlib.metadata

### PR DESCRIPTION
**Pull request by mthuurne**
Until now, pydoctor used Incremental to define the version in code and extract it from there during the build to insert it into the
metadata. This commit reverses that approach, by defining the version in metadata and having the code extract it from there.

Static metadata is easier to use by tools such as pydoctor itself. In the future we'll probably want to adopt PEP 621 for that purpose; this is a stepping stone.

### CODE REVIEW

**Review by adiroiban — APPROVED** [review]
Looks good. Happy to see that "incremental" is removed :)

File: pydoctor/__init__.py [hunk 1]
```diff
@@ -4,10 +4,17 @@
...
+# On older versions, a compatibility package can be installed from PyPI.
+try:
+    import importlib.metadata as importlib_metadata
+except ImportError:
+    if not TYPE_CHECKING:
+        import importlib_metadata
 
-from ._version import __version__
 
+__version__ = importlib_metadata.version('pydoctor')
```
Comment (graingert) [inline pydoctor/__init__.py:18]: changing this from an Incremental instance to a str is a breaking change - I think you might as well forgo the `__version__ =` entirely and require callers `importlib.metadata.version("pydoctor")` if they ever forget what version they installed

Comment (mthuurne) [reply pydoctor/__init__.py:18]: That's a fair point, but pydoctor doesn't offer a stable API yet, so there shouldn't be many callers. Even Twisted's customizations could be phased out; see #315.

**Comment by mthuurne** [comment]
Note that we do have to update Twisted's unit tests to not access `__version__`, or at least not expect there to be an Incremental `Version` object there. But that doesn't need to happen until we do a new major release, which might have other breaking changes as well.

**Pull request merged**